pkg/agent/plugin/nodeattestor/k8s/psat/psat_test.go

```diff
@@ -0,0 +1,234 @@
+package psat
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/spiffe/spire/pkg/common/pemutil"
+	sat_common "github.com/spiffe/spire/pkg/common/plugin/k8s"
+	"github.com/spiffe/spire/proto/agent/nodeattestor"
+	"github.com/spiffe/spire/proto/common/plugin"
+	"github.com/stretchr/testify/suite"
+	jose "gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+var sampleKeyPEM = []byte(`-----BEGIN RSA PRIVATE KEY-----
+MIIBywIBAAJhAMB4gbT09H2RKXaxbu6IV9C3WY+pvkGAbrlQRIHLHwV3Xt1HchjX
+c08v1VEoTBN2YTjhZJlDb/VUsNMJsmBFBBted5geRcbrDtXFlUJ8tQoQx1dWM4Aa
+xcdULJ83A9ICKwIDAQABAmBR1asInrIphYQEtHJ/NzdnRd3tqHV9cjch0dAfA5dA
+Ar4yBYOsrkaX37WqWSDnkYgN4FWYBWn7WxeotCtA5UQ3SM5hLld67rUqAm2dLrs1
+z8va6SwLzrPTu2+rmRgovFECMQDpbfPBRex7FY/xWu1pYv6X9XZ26SrC2Wc6RIpO
+38AhKGjTFEMAPJQlud4e2+4I3KkCMQDTFLUvBSXokw2NvcNiM9Kqo5zCnCIkgc+C
+hM3EzSh2jh4gZvRzPOhXYvNKgLx8+LMCMQDL4meXlpV45Fp3eu4GsJqi65jvP7VD
+v1P0hs0vGyvbSkpUo0vqNv9G/FNQLNR6FRECMFXEMz5wxA91OOuf8HTFg9Lr+fUl
+RcY5rJxm48kUZ12Mr3cQ/kCYvftL7HkYR/4rewIxANdritlIPu4VziaEhYZg7dvz
+pG3eEhiqPxE++QHpwU78O+F1GznOPBvpZOB3GfyjNQ==
+-----END RSA PRIVATE KEY-----`)
+
+func TestAttestorPlugin(t *testing.T) {
+	suite.Run(t, new(AttestorSuite))
+}
+
+type AttestorSuite struct {
+	suite.Suite
+
+	dir      string
+	attestor *nodeattestor.BuiltIn
+}
+
+func (s *AttestorSuite) SetupTest() {
+	var err error
+	s.dir, err = ioutil.TempDir("", "spire-k8s-psat-test-")
```

Comment:
hmm, this is my fault, but the temp directory isn't being cleaned up. Do you mind adding a TearDownTest method on this suite that removes the temp directory? (and fix the SAT test too)